### ISSUE TITLE
Seeking guidance on 2.2.1 Timing Adjustable and session timeouts

### ISSUE THREAD

**Issue by jamesjacobs** [w3c/wcag#3206]
Would really appreciate some clarification on form session timeouts and 2.2.1 Timing Adjustable (A). 

Consider the following scenario:

Say I have a multi-page form that collects personal details. The server session timeout limit is set at 20 minutes. Each time the user submits a page in the form or navigates back and forth throughout the form, the session timeout limit refreshes. If a user takes longer than 20 mins to complete a page, then hits submit, they are sent back to the start of the form and all data previously entered is lost. There is no option to extend the session, and no warning that the session is about to expire. There is, however, instructions on the first page of the form advising the user that they have no more than 20 minutes to complete each page or progress will be lost.

It is my understanding that this would fail 2.2.1 as none of the 6 options in the SC are met.

That being said, I do wonder if having the timeout warning on the first page, and the action of clicking back and forth throughout the form via next/back buttons (which refreshes the timeout limit via a page reload) is a "simple action" and therefore does meet the **Extend**  SC???

I also see this sentence under 2.2.6:
> However, when it is not practical to save the user data the author must warn the user about the duration of inactivity which will result in a timeout.

While this is obviously a different SC, it's closely related to session timeouts.

I fully understand that a more desirable approach would perhaps be an explicit prompt to the user, in the UI, advising of the impending timeout with the option of extending the session but would really appreciate feedback on the above.

Thanks in advance.

**Comment by alastc** [comment]
I think this snippet from the [understanding document](https://w3c.github.io/wcag/understanding/timing-adjustable) is the answer to your question:

> Time limits set externally to content, such as by the user agent or by factors intrinsic to the Internet are not under the author's control and not subject to WCAG conformance requirements. Time limits set by Web servers should be under the author's/organization's control and are covered.

It sounds like your case is the latter, under the control of the 'content'. 

The SC is trying to avoid the situation you outline:
> If a user takes longer than 20 mins to complete a page, then hits submit, they are sent back to the start of the form and all data previously entered is lost. There is no option to extend the session, and no warning that the session is about to expire.

If there is enough functionality to extend the session on each submit/navigation, there should be the capability to create a warning and extension before it times out.

2.2.6 is related, but it's text does not apply to other guidelines.

Please close the issues if this answers your question.

**Comment by jamesjacobs** [comment]
Thank you for the response @alastc
